titiler/pgstac/factory.py

```diff
@@ -536,7 +546,42 @@ def wmts(
                 "format": tile_format.value,
                 "tileMatrixSetId": tileMatrixSetId,
             }
-            tiles_url = self.url_for(request, "tile", **route_params)
+
+            # `route_params.copy()` this can be removed after titiler>=0.13.2 update
+            tiles_url = self.url_for(request, "tile", **route_params.copy())
+
+            # List of dependencies a `/tile` URL should validate
+            # Note: Those dependencies should only require Query() inputs
+            tile_dependencies = [
+                self.layer_dependency,
+                self.dataset_dependency,
+                self.pixel_selection_dependency,
+                self.process_dependency,
+                self.rescale_dependency,
+                self.colormap_dependency,
+                self.render_dependency,
+                PgSTACParams,
+                self.reader_dependency,
+            ]
+
+            layers: List[Dict[str, Any]] = []
+            if search_info.metadata.defaults:
+                for name, values in search_info.metadata.defaults.items():
+                    query_string = urlencode(values, doseq=True)
+                    try:
+                        self.check_query_params(
+                            dependencies=tile_dependencies,
+                            query_params=QueryParams(query_string),
+                        )
+                    except Exception:
+                        continue
```

Comment:
we should maybe log the exception for debugging purpose. we could also run this in the `/register` endpoint to validate `defaults` metadata